### PR TITLE
Cleaning June 1

### PR DESCRIPTION
**Pull request by mrshirts**
A number of things here:

- fixed bugs so that parallel simulations can run run_replica_exchange.py using mpirun.  Call is
   `mpirun -np [number of processes] ./run_replica_exchange.py1
-in order to make it run in parallel, separated out run_replica_exchange.py and process_replica_exchange.py.  One runs run_replica_exchange.py, then afterwards runs process_replica_exchange.py.  I had to add a kludge where it pickles some of the information needed to process the files; this should be improved eventually.
- torsion parameters should have the correct units now.
- removed Yank dependencies.
- replica files are written inside the output directory
- tried to reduce passing information if it was already stored in the simulation output.
- automatically minimized the run_replica_exchange.py runs (optionally) and the simulation.py runs
- removed some of the error handling for bad timesteps (things that only needed to be run once or twice)
- removed some .pyc files that didn't need to be in git.
-constraints seem not to work in general.  It will take some checking later.
- the structure is a bit different now.

### CODE REVIEW

File: examples/replica_exchange/process_replica_exchange.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+from simtk import unit
+from foldamers.cg_model.cgmodel import CGModel
+from foldamers.parameters.reweight import get_temperature_list
+from cg_openmm.simulation.rep_exch import *
+import numpy as np
+import pickle
+
+###
+#
+# This example demonstrates how to run a Yank replica exchange simulation
```
Comment (cwalker7) [inline examples/replica_exchange/process_replica_exchange.py:14]: Get rid of 'Yank' in comment

File: examples/replica_exchange/process_replica_exchange.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+#
+###
+
+# Job settings
+output_directory = "output"
+if not os.path.exists(output_directory):
+    os.mkdir(output_directory)
+output_data = os.path.join(output_directory, "output.nc")
+
+# Replica exchange simulation settings
```
Comment (cwalker7) [inline examples/replica_exchange/process_replica_exchange.py:25]: Do these min and max temps need to be defined here?

File: examples/replica_exchange/run_replica_exchange.py [hunk 3]
```diff
@@ -85,7 +88,7 @@
 }
 
 # Torsion angle definitions
-torsion_force_constant = 0.2 * unit.kilocalorie_per_mole / unit.radian / unit.radian
+torsion_force_constant = 20.0 * unit.kilocalorie_per_mole / unit.radian / unit.radian
```
Comment (cwalker7) [inline examples/replica_exchange/run_replica_exchange.py:91]: Periodic torsion units should be kcal/mol?

File: examples/replica_exchange/process_replica_exchange.py (hunk 2 above)
Comment (mrshirts) [reply examples/replica_exchange/process_replica_exchange.py:25]: No, passed in the pkl for now.

File: examples/replica_exchange/process_replica_exchange.py (hunk 1 above)
Comment (mrshirts) [reply examples/replica_exchange/process_replica_exchange.py:14]: fixed.

File: examples/replica_exchange/run_replica_exchange.py (hunk 3 above)
Comment (mrshirts) [reply examples/replica_exchange/run_replica_exchange.py]: Yep, didn't fix it here.

**Pull request merged**